operator/autoscaler_test.go

```diff
@@ -341,15 +341,36 @@ func TestScaleUpCausedByShardToNodeRatioExceeded(t *testing.T) {
 	require.Equal(t, UP, actual.ScalingDirection, actual.Description)
 }
 
+func TestScaleDownCausedByShardToNodeRatioLessThanOne(t *testing.T) {
+	eds := edsTestFixture(12)
+	esNodes := make([]ESNode, 0)
+
+	// require to scale-up because we exceeded shard-to-node ratio limits.
+	eds.Spec.Scaling.MaxShardsPerNode = 6
+	eds.Spec.Scaling.MaxReplicas = 999
+	esIndices := map[string]ESIndex{
+		"ad1": {Replicas: 1, Primaries: 3, Index: "ad1"},
+	}
+	// scaling independent of desired scaling direction
+	scalingHint := UP
+
+	as := systemUnderTest(eds, nil, nil)
+
+	actual := as.calculateScalingOperation(esIndices, esNodes, scalingHint)
+	require.Equal(t, int32(12), *actual.NodeReplicas, actual.Description)
+	require.Equal(t, 1, len(actual.IndexReplicas), actual.Description)
+	require.Equal(t, UP, actual.ScalingDirection, actual.Description)
+}
+
 func TestAtMaxShardsPerNode(t *testing.T) {
 	eds := edsTestFixture(4)
 	esNodes := make([]ESNode, 0)
 
 	// don't scale down if that would violate the maxShardsPerNode
-	eds.Spec.Scaling.MaxShardsPerNode = 6
+	eds.Spec.Scaling.MaxShardsPerNode = 12
 	esIndices := map[string]ESIndex{
-		"ad1": {Replicas: 1, Primaries: 9, Index: "ad1"},
-		"ad2": {Replicas: 0, Primaries: 5, Index: "ad2"},
+		"ad1": {Replicas: 1, Primaries: 18, Index: "ad1"},
```

Comment:
this test unveils an interesting difference in behaviour that we should capture as test. Previously it seems that scaling down didn't align the index replicas like it does now.

Reply:
I think I've already added test for this
https://github.com/zalando-incubator/es-operator/pull/157/files#diff-79598d30b7d766804171bf9a57fb7b2833b1f3e3ca7073e083d43fe8417477d4R385

ScalingHint=Down but we expect increasing number of replicas and an actual behavior is scaling UP

Reply:
Here we have an edge-case situation of index replicas being different within the EDS. I think this deserves its own test. The other test is fine. 

Can you also add a comment with the math why we'd be exceeding max-shards per node?

Reply:
> Here we have an edge-case situation of index replicas being different within the EDS. I think this deserves its own test. The other test is fine.

Sorry, I didn't get this. Could you please explain it or provide some example?

Reply:
all good, I think we got all covered now.